OutOfSchool/OutOfSchool.WebApi/Services/AreaAdminService.cs

```diff
@@ -519,7 +506,11 @@ public async Task<bool> IsAreaAdminSubordinateRegionCreateAsync(string regionAdm
             predicate = predicate.And(a => a.Institution.Id == filter.InstitutionId);
         }
 
-        if (filter.CATOTTGId > 0)
+        if (catottgs.Count > 0 && catottgs.Contains(filter.CATOTTGId))
+        {
+            predicate = predicate.And(a => a.Institution.Id == filter.InstitutionId && a.CATOTTG.Id == filter.CATOTTGId);
```

Comment:
Please refactor filtering logic.

What if `filter.InstitutionId == Guid.Empty` but `catottgs.Count > 0 && catottgs.Contains(filter.CATOTTGId)` is true? 
I see that `GetByFilter()` prevents this now, but there are no obstacles for it to be changed later. And then you'll get nothing returned.

What if `filter.InstitutionId != Guid.Empty` and `catottgs.Count > 0 && catottgs.Contains(filter.CATOTTGId)` is true ?
Then condition `a.Institution.Id == filter.InstitutionI` would be applied twice, once in the previos `if`, and second time here. 

As I can't say if double applying of same condition would be coalesced or not, I'd suggest to avoid doing it.